### PR TITLE
Migrate frontend from jQuery to Vue.js

### PR DESCRIPTION
**Pull request by merlinran**
This PR also includes https://github.com/Twisted-Fields/acorn-precision-farming-rover/pull/4 so can be reviewed after that, but it's also okay to just review and merge this.

Main changes:
1. Upgrades bootstrap to v5 to eliminate the dependency to jQuery.
2. Replace anything jQuery does to Vue.js.

### CODE REVIEW

File: server/server.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def robots_to_json(keys):
         robot = pickle.loads(redis_client.get(key))
         if robot is None:
             print("Database returned None for key {}".format(key))
-        time_stamp = json.dumps(robot.time_stamp, default=date_handler)
+        time_stamp = date_handler(robot.time_stamp)
```
Comment (merlinran) [inline server/server.py:96]: `json.dumps` added an extra pair of double quotes to the final JSON, which did nothing but adding complexity.

File: server/static/js/actions.js [hunk 2]
```diff
@@ -0,0 +1,607 @@
+// actions are triggered by UI events. It mostly involves calling backend API, and as a result, updating the store.
```
Comment (merlinran) [inline server/static/js/actions.js:1]: This file includes the functions interacting with server or drawing graphs originally in `server/static/js/acorn_leaflet.js`.

File: server/static/js/components.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
+Vue.component("robot-cardview", {
```
Comment (merlinran) [inline server/static/js/components.js:1]: This is the code managing the right side panel showing robot information and control buttons.

File: server/static/js/components.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    },
+    gpsAction: function (event) {
+      const cmd = event.target.innerText;
+      this.currentGPSCommand = cmd;
+      updateGpsRecordCommand(this.robot.name, cmd);
+    },
+  },
+});
+
+Vue.component("control-panel", {
```
Comment (merlinran) [inline server/static/js/components.js:45]: This is the top panel

File: server/static/js/view.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
+// http://192.168.1.170:8090/api/projects/3/tasks/3116cce4-4215-4de9-9e9a-0e9c93df87f6/orthophoto/tiles.json
```
Comment (merlinran) [inline server/static/js/view.js:1]: This file contains the rest of the original `server/static/js/acorn_leaflet.js`

File: server/templates/control-panel.html [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    <div class="col-sm-3">
+      <h1>Acorn control</h1>
+    </div>
+
+    <div class="col-md-5">
+      <div class="input-group" id="dropdown-container">
+        <button
+          type="button"
+          class="btn btn-info mr-1"
+          @click="toggleCollapsePlots"
```
Comment (merlinran) [inline server/templates/control-panel.html:17]: This equals to `v-on:click`, which calls the corresponding methods defined in `components.js`.

**Comment by tlalexander** [comment]
This sounds great. I am looking forward to seeing what this looks like when I have some time.

**Comment by tlalexander** [comment]
I've merged this but I just realized that the dropdown on the web page isn't loading properly. I am reading about vue.js and how it works, and it seems like you've plumbed it in to be bound to the pathNames variable. And yet, the dropdown never populates with names for me. Would you be able to take a look? It's on the current main branch. I may have messed something up with the merge. Thank you!

**Comment by merlinran** [comment]
Oh I think I know why. `server/static/js/lib/bootstrap.bundle.min.js` and `server/static/js/lib/bootstrap.bundle.min.js.map` exist in my branch but are missing from `main`. The dropdown menu is the only place bootstrap.bundle.min.js is still used, but since it works well, and we rely on bootstrap for styling anyway, so I didn't change it to some vue.js plugin.

I'll create a quick PR to add the two files back.

**Comment by merlinran** [comment]
Verified locally that the dropdown works with https://github.com/Twisted-Fields/acorn-precision-farming-rover/pull/9.

**Comment by tlalexander** [comment]
Thank you! That fix works great.

**Pull request merged**